File: programmers/코딩테스트연습/정렬/H-index.py
def solution(citations):
    answer = []
    for h in range(max(citations)):
        cnt = 0
        for cite in citations:
            if h <= cite:
                cnt += 1
        if cnt >= h:
            answer.append(h)
    return max(answer) 

if __name__ == "__main__":
    print(solution( [3, 0, 6, 1, 5] ))

File: programmers/코딩테스트연습/정렬/H-index.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> citations) {
    if(citations.size() == 1){
        return 1;
    } else {
        sort(citations.begin(), citations.end());
        int answer = 0;
        int cSize = citations.size();
        for(int j = 0; j < cSize; j++){
            int h = citations[j];
            int count = cSize - j;
            if(count <= h){
                answer = count;
                break;
            }
        }
        return answer;
    }
}